src/script/standard.cpp

```diff
@@ -89,21 +89,49 @@ static constexpr bool IsSmallInteger(opcodetype opcode)
     return opcode >= OP_1 && opcode <= OP_16;
 }
 
+/** Valid number of keys in a multisig are [1,20]. */
+static constexpr bool IsValidKeysNumber(unsigned int n_keys)
+{
+    return n_keys > 0 && n_keys <= MAX_PUBKEYS_PER_MULTISIG;
+}
+
 static bool MatchMultisig(const CScript& script, unsigned int& required, std::vector<valtype>& pubkeys)
 {
     opcodetype opcode;
     valtype data;
+    unsigned int keys;
+
     CScript::const_iterator it = script.begin();
     if (script.size() < 1 || script.back() != OP_CHECKMULTISIG) return false;
 
-    if (!script.GetOp(it, opcode, data) || !IsSmallInteger(opcode)) return false;
-    required = CScript::DecodeOP_N(opcode);
+    if (!script.GetOp(it, opcode, data)) return false;
+    if (IsSmallInteger(opcode)) {
+        required = CScript::DecodeOP_N(opcode);
+    } else {
+        try {
+            required = CScriptNum(data, true).getint();
+        } catch (const scriptnum_error&) {
+            return false;
+        }
+    }
```

Comment:
This wants to be a function of its own I think :)

Reply:
Yep :) Done